### ISSUE TITLE
use a crash report/logging provider or library

### ISSUE THREAD

**Issue by Trolldemorted** [signal-csharp/Signal-Windows#105]
A quick google search yields services like HockeyApp or the Mobile Center SDK. Does anyone have experience with these, or similar solutions?

cc @golf1052, @SERVCUBED

**Comment by SERVCUBED** [comment]
I agree we definitely need some form of telemetry. I have had some experience with Microsoft's Application Insights and I'm looking into HockeyApp now.

**Comment by Trolldemorted** [comment]
I have added basic logging to the most frequent code paths in libsignal-service-dotnet, and we can add handlers to it in Signal-Windows. Right now one is printing every log event, but it should be easy to add ring-buffered logs in memory or to save them on disk.

**Comment by Trolldemorted** [comment]
I tested AppCenterSDK, but their [crash reports don't work reliably](https://github.com/Microsoft/AppCenter-SDK-DotNet/issues/572) and they do not (yet) support attaching logfiles.
